runtimes/go/appruntime/apisdk/api/transport/eh2c.go

```diff
@@ -0,0 +1,35 @@
+package transport
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"net/http"
+
+	"golang.org/x/net/http2"
+)
+
+func NewH2CTransport() http.RoundTripper {
+	return &H2CTransport{
+		h2c: &http2.Transport{
+			AllowHTTP: true,
+			DialTLSContext: func(ctx context.Context, network, addr string, _ *tls.Config) (net.Conn, error) {
+				var d net.Dialer
+				return d.DialContext(ctx, network, addr)
+			},
+		},
+	}
+}
+
+type H2CTransport struct {
+	h2c http.RoundTripper
+}
+
+func (h H2CTransport) RoundTrip(request *http.Request) (*http.Response, error) {
```

Comment:
Since we're returning a pointer from `NewH2CTransport, it's clearer if the method is on a pointer too

```suggestion
func (h *H2CTransport) RoundTrip(request *http.Request) (*http.Response, error) {
```